### PR TITLE
🐛  Use ptr instead of own implementation

### PR DESCRIPTION
**Pull request by sivchari**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
/ok-to-test

Thanks for this!

File: errors/pointer.go [hunk 1]
```diff
@@ -14,6 +14,10 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// Deprecated: errors returns pointer of type received as input, but it can be satisfied by using ptr package.
+// It keeps these functions for backward compatibility.
+//
+// cluster-api no longer uses this package, and it will be removed in the future, so please use the ptr package instead.
```
Comment (killianmuldoon) [inline errors/pointer.go:20]:
```suggestion
```

Comment (killianmuldoon) [reply errors/pointer.go:20]: We can't deprecate the package as there are other functions in it which are not deprecated.

File: errors/pointer.go [hunk 2]
```diff
@@ -14,6 +14,10 @@ See the License for the specific language governing permissions and
...
 package errors
 
 // MachineStatusErrorPtr converts a MachineStatusError to a pointer.
```
Comment (killianmuldoon) [inline errors/pointer.go:23]: Let's add a deprecation message to each of the individual functions.

The usual form is:

```
// Deprecated: This function is deprecated and will be removed in an upcoming release of Cluster API.
```

File: errors/pointer.go (hunk 1 above)
Comment (sivchari) [reply errors/pointer.go:20]: I narrowed the deprecated scope, thanks.

**Comment by sivchari** [comment]
@killianmuldoon 
I changed them and would like you to review them again.

**Review by killianmuldoon — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**